src/main/java/cmput402/tdd/App.java

```diff
@@ -158,6 +158,80 @@ public void editShoppingList(Scanner input, ShoppingList shoppingList){
         }
 
     }
+    public void editRecipe(Scanner input, Recipe recipe) throws Exception{
+        clearConsole();
+        System.out.println("Please select an option to edit your recipe:");
+        System.out.println("command format (ignore bracket): (add 1) (remove 5) (rename \"apples\")");
+        System.out.println("*add\n*remove\n*rename\n");
+        Integer quantity;
+        Item item;
+        String param;
+        String command = input.nextLine();
+        /*
+        Eg:
+        command = add 5
+
+        function will split command into variables action and param
+        action = add
+        param = 5
+         */
+        Integer spaceIndex = command.indexOf(" ");
+        if(spaceIndex == -1){
+            throw new Exception("Invalid command format");
+        }
+        String action = command.substring(0, spaceIndex);
+
+        if(action.equals("add") || action.equals("remove")) {
+            param = command.substring(spaceIndex + 1, command.length());
+            try{
+                quantity = Integer.parseInt(param);
+            }
+            catch(Exception e){
+                throw new Exception("Please enter a positive whole number for quantity");
+            }
+            try{
+                item = createItem(input);
+            }
+            catch (Exception e) {
+                throw new Exception("Failed to create item for recipe");
+            }
+            if(action.equals("add")){
+                if(quantity > 0){
+                    recipe.add(item, quantity);
+                    System.out.println("Added "+param+" "+item.getName()+" to recipe successfully");
+                }
+                else{
+                    throw new Exception("Could not add item to recipe");
+                }
+
+            }
+            else if(action.equals("remove")){
+                if(recipe.items.get(item) > quantity && quantity > 0){
```

Comment:
Java short circuit evaluates left to right, checking if the quantity is above 0 first will save you the check of if the quantity is less than what is there

Reply:
Changed in latest commit.

Reply:
After investigating, I put the quantity > 0 test into the code because it is needed in order for conditions that do not meet the condition to reach the else statement where it throws an exception, if I were to move the quantity > 0 condition to the outside else if statement, it would require me to repeat 
`else{
                    throw new Exception("Could not remove item from recipe");
                }`
Which is redundant, the current implementation is best.

Reply:
No I meant 
```suggestion
                if(quantity > 0 && recipe.items.get(item) > quantity){
```

Reply:
I understand now, made suggest change.